### ISSUE TITLE
Licensing as GPLv2+?

### ISSUE THREAD

**Issue by JeremyRand** [JesusMcCloud/netlayer#8]
Hello,

I figure I should probably mention this now while there are only 2 contributors, and it's relatively easy to handle licensing issues.  Right now there seems to be some legal uncertainty with the EUPL's compatibility with GPL.  Wikipedia claims that EUPL is compatible with GPL, while FSF claims that it is not compatible; I suspect that the reality is slightly more convoluted than either of those opinions.

Is there any chance you might be willing to explicitly license netlayer under GPLv2+ (or maybe GPLv3+ if you prefer that) in addition to the current EUPL1.1+ license, so that developers who wish to utilize netlayer in GPL projects don't have to worry about legal uncertainties?  (This is made more important by the fact that many GPL projects -- including the ones I work with -- don't have budget for asking lawyers about things, so any legal uncertainties, even ones that are ultimately baseless, are a roadblock for those projects.)

Cheers!

**Comment by JesusMcCloud** [comment]
Sorry for the late reply!
I fully understand your situation. Where does the FSF state that the EUPL is not compatible with the GPL? The body who created the EUPL even states that it is GPL-compatible (see https://joinup.ec.europa.eu/page/eupl-compatible-open-source-licences).

**Comment by JeremyRand** [comment]
> Where does the FSF state that the EUPL is not compatible with the GPL?

https://www.gnu.org/licenses/license-list.html#EUPL

From what I gather by reading between the lines (which I'm probably not supposed to do as a non-lawyer), the issue seems to be that EUPLv1.1 allows relicensing to GPLv2, and EUPLv1.2 allows relicensing to GPLv2 and GPLv3, but neither EUPL version allows relicensing to *future* GPL versions.  Since a lot of GPL projects follow FSF's recommendation to allow relicensing to future GPL versions (i.e. GPLv2+ and GPLv3+ both allow relicensing to GPLv4), in practice EUPL-licensed code can't be integrated into those GPL projects.  (It is likely that future EUPL versions would allow relicensing to any future GPL versions that have been released by that point, but such a likelihood isn't legally sufficient since it's not certain.)

**Comment by JesusMcCloud** [comment]
> TL;DR: wontfix

Thanks for the clarification!
I am skeptical when it comes to permitting _future_ GPL versions. Of course, you are right! EUPL is compatible to GPLv2 and GPLv3 and in my mind that was the end of it. However, I can think of a couple of (far more) important projects which also do not permit re-licensing to future GPL versions and still do fine.
Therefore, my answer for now is that this is not going to happen in the foreseeable future.
